### PR TITLE
Test PossibleFragmentSpreadsRule

### PR DESCRIPTION
**Pull request by PascalSenn**
Ports validation tests from GraphQLJs for rule PossibleFragmentSpreadsRule

### CODE REVIEW

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 1]
```diff
@@ -58,21 +48,13 @@ public void InterfaceTypeDoesMatch()
...
-
-            // act
-            Rule.Validate(context, query);
-
-            // assert
-            Assert.Empty(context.Errors);
         }
 
         [Fact]
         public void UnionTypeDoesMatch()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:54]: @michaelstaib  Can this really work?

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 2]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                    }
+                }
+
+                fragment invalidObjectWithinUnion on CatOrDog { ...humanFragment }
+                fragment humanFragment on Human { pets { name } }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoNotContainedObject()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:371]: @michaelstaib  This one should fail but passes

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 3]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                        }
+                    }
+                }
+                fragment interfaceWithinInterface on Pet { ...beingFragment }
+                fragment beingFragment on Being { name }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoOverlappingInterfaceInInlineFragment()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:272]: @michaelstaib  is this an interface implement interface on?

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 4]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                        ...interfaceWithinObject
+                    }
+                }
+                fragment interfaceWithinObject on Dog { ...petFragment }
+                fragment petFragment on Pet { name }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoOverlappingInterface()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:255]: @michaelstaib  is this an interface implement interface on?

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 5]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                        ...invalidUnionWithinObject
+                    }
+                }
+                fragment invalidUnionWithinObject on Human { ...catOrDogFragment }
+                fragment catOrDogFragment on CatOrDog { __typename }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoNonOverlappingInterface()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:386]: @michaelstaib This one should fail but passes

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 6]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                        }
+                    }
+                }
+                fragment invalidUnionWithinInterface on Pet { ...humanOrAlienFragment }
+                fragment humanOrAlienFragment on HumanOrAlien { __typename }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoNonOverlappingUnion()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:403]: @michaelstaib This one should fail but passes

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs [hunk 7]
```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
...
+                    }
+                }
+                fragment invalidInterfaceWithinInterfaceAnon on Pet {
+                    ...on Intelligent { iq }
+                }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoNonOverlappingUnion()
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:470]: @michaelstaib This one should fail but passes

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs (hunk 1 above)
Comment (michaelstaib) [reply src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:54]: This one ... yes since pet implements dog.

Comment (michaelstaib) [reply src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:54]: I mean dog implements pet :)

File: src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs (hunk 3 above)
Comment (michaelstaib) [reply src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs:272]: yes

**Pull request merged**